add edit article endpoint

vorobevamv opened the pull request:
added edit article endpoint;
added EF Support

Commit:
EditArticleRequestValidator

Commit:
EditArticleRequestValidator (currentRubricId + currentArticleName)

Commit:
deleted double foreach

Commit:
to update develop

Commit:
resolve merge conflicts

Commit:
added edit article endpoint;
added EFSupport

MariyaSin commented on src/WikiService.Mappers/Models/PatchDbArticleMapper.cs:
```diff
@@ -0,0 +1,28 @@
+        return null;
+      }
+
+      var result = new JsonPatchDocument<DbArticle>();
```
please, do not use var

vorobevamv replied:
fixed

Commit:
removed var in Mapper

Commit:
changes in EditArticleRequestValidator

Commit:
in EditAsync removed receiving ArticleDb;
in EditArticleRequestValidator removed check to Id format

Commit:
changes in EditArticleRequestValidator

Commit:
changes in EditArticleRequestValidator

Commit:
changes in EditArticleRequestValidator

VLiamin commented on src/WikiService.Data/ArticleRepository.cs:
```diff
@@ -1,21 +1,26 @@
     public ArticleRepository(
-      IDataProvider provider)
+      IDataProvider provider,
+       IHttpContextAccessor httpContextAccessor)
```
space

vorobevamv replied:
fixed

VLiamin commented on src/WikiService.Data/ArticleRepository.cs:
```diff
@@ -37,9 +42,24 @@ public async Task<bool> DoesSameNameExistAsync(Guid rubricId, string articleName
+
+    public async Task<bool> EditAsync(DbArticle dbArticle, JsonPatchDocument<DbArticle> request)
+    {
+      if (dbArticle == null || request == null)
```
if (dbArticle is null || request is null)

vorobevamv replied:
done

VLiamin commented on src/WikiService.Mappers/Models/PatchDbArticleMapper.cs:
```diff
@@ -0,0 +1,28 @@
+
+      JsonPatchDocument<DbArticle> result = new JsonPatchDocument<DbArticle>();
+
+      foreach (var item in request.Operations)
```
May be replace var?

vorobevamv replied:
done

VLiamin commented on src/WikiService.Mappers/Models/PatchDbArticleMapper.cs:
```diff
@@ -0,0 +1,28 @@
+  {
+    public JsonPatchDocument<DbArticle> Map(JsonPatchDocument<EditArticleRequest> request)
+    {
+      if (request == null)
```
is null

vorobevamv replied:
done

VLiamin commented on src/WikiService.Validation/Article/EditArticleRequestValidator.cs:
```diff
@@ -0,0 +1,126 @@
+
+             return (_currentRubricId == x.Item1.RubricId && _currentArticleName == x.Item1.Name)
+             ? true
+             :!await _articleRepository.DoesSameNameExistAsync(_currentRubricId, _currentArticleName);
```
space

vorobevamv replied:
fixed

VLiamin commented on src/WikiService.Validation/Article/EditArticleRequestValidator.cs:
```diff
@@ -0,0 +1,126 @@
+  {
+    private readonly IArticleRepository _articleRepository;
+    private readonly IRubricRepository _rubricRepository;
+    private async Task HandleInternalPropertyValidationAsync(Operation<EditArticleRequest> requestedOperation, CustomContext context)
```
empty line

vorobevamv replied:
fixed

VLiamin commented:
May be change name of this PR to  Add edit article endpoint? Without feature?

Commit:
changes according comments

DarkFelas commented on src/WikiService.Models.Dto/Requests/Article/EditArticleRequest.cs:
```diff
@@ -0,0 +1,12 @@
+
+namespace LT.DigitalOffice.WikiService.Models.Dto.Requests.Article
+{
+  public record EditArticleRequest
```
There is no need in record if u don't use its features :)
Whole this text can be simplified to:


```
using System;

namespace LT.DigitalOffice.WikiService.Models.Dto.Requests.Article;

public record EditArticleRequest(string Name, string Content, bool IsActive, Guid RubricId);
```

vorobevamv replied:
OK, fixed. thanks)

Commit:
fixed EditArticleRequest;
added "StringComparison.OrdinalIgnoreCase" to validator;
IsNullOrEmpty+Trim changed to IsNullOrWhiteSpace;

MariyaSin commented on Common.Configurations.targets:
```diff
@@ -25,16 +25,10 @@
-    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.5" />
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.61" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.6" />
+	<PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
```
fix tab

MariyaSin commented on src/WikiService.Validation/Article/EditArticleRequestValidator.cs:
```diff
@@ -0,0 +1,129 @@
+        {
+          {
+            async x => await _rubricRepository.DoesExistAsync(
+            Guid.TryParse(x.value?.ToString(), out Guid _rubricId) ? _rubricId : default
```
add tab

MariyaSin commented on src/WikiService.Validation/Article/EditArticleRequestValidator.cs:
```diff
@@ -0,0 +1,129 @@
+             }
+
+             return (_currentRubricId == x.Item1.RubricId && _currentArticleName == x.Item1.Name)
+             ? true
```
add tab

MariyaSin commented on src/WikiService.Validation/Article/EditArticleRequestValidator.cs:
```diff
@@ -0,0 +1,129 @@
+
+             return (_currentRubricId == x.Item1.RubricId && _currentArticleName == x.Item1.Name)
+             ? true
+             : !await _articleRepository.DoesSameNameExistAsync(_currentRubricId, _currentArticleName);
```
add tab

Commit:
EditArticleRequest: changed record to class + changes according comments

Commit:
fix tabs

Commit:
tabs
